patches/minecraft/net/minecraft/item/ItemStack.java.patch

```diff
@@ -284,4 +284,18 @@
 +            return this.capabilities.areCompatible(other.capabilities);
 +        }
 +    }
++
++    public static boolean areItemStacksWithShareTagEqual(@Nullable ItemStack stackA, @Nullable ItemStack stackB)
++    {
++        if (stackA == null && stackB == null) return true;
++        if (stackA == null && stackB != null) return false;
++        if (stackA != null && stackB == null) return false;
++        if (stackA.field_77994_a != stackB.field_77994_a) return false;
++        if (stackA.field_151002_e != stackB.field_151002_e) return false;
++        if (stackA.field_77991_e != stackB.field_77991_e) return false;
++        if (stackA.field_151002_e == null) return false;
++        if (stackA.field_151002_e.getNBTShareTag(stackA) == null) return false;
```

Comment:
Good catch, thanks.
I guess the context in which I am using this new method would never trigger it with nbtsharetag being null for stackB (would actually never trigger any of the previous conditions either), but I will fix this so that it really does what the name says.